### PR TITLE
MULE-19229: Remove lazyInit concept from MuleConfiguration

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.size.SmallTest;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+@SmallTest
+public class MuleConfigurationTestCase extends AbstractMuleTestCase {
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java:39]: Add allure annotations

File: core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    MuleContextBuilder contextBuilder = MuleContextBuilder.builder(APP);
+    contextBuilder.setMuleConfiguration(config);
+    muleContext = new DefaultMuleContextFactory()
+        .createMuleContext(asList(testServicesConfigurationBuilder, new DefaultsConfigurationBuilder()), contextBuilder);
+    muleContext.start();
+
+    verifyConfiguration();
+  }
+
+  /** Test for MULE-3092 */
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java:88]: Use allure annotations and @Issue to refer to the corresponding ticket.

File: core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    testServicesConfigurationBuilder.registerAdditionalService("mockedConfigurationProperties", mockedConfigurationProperties);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    muleContext.dispose();
+    muleContext = null;
+  }
+
+  /** Test for MULE-3092 */
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java:61]: Use allure annotations and @Issue to refer to the corresponding ticket.

File: core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "serverId");
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "domainId");
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "message.cacheBytes");
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "message.cacheOriginal");
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "streaming.enable");
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "message.assertAccess");
+    clearProperty(SYSTEM_PROPERTY_PREFIX + "transform.autoWrap");
+  }
+
+  /** Test for MULE-3110 */
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java:132]: Use allure annotations and @issue to refer to the corresponding ticket.

File: core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    assertTrue(config.isClientMode());
+
+    // These are not OK to change after init
+    assertFalse("UTF-16".equals(config.getDefaultEncoding()));
+    assertFalse(workingDirectory.equals(config.getWorkingDirectory()));
+    assertFalse("MY_SERVER".equals(config.getId()));
+    assertFalse("MY_DOMAIN".equals(config.getDomainId()));
+  }
+
+  /** Test for MULE-3110 */
```
Comment (fsgonz) [inline core-tests/src/test/java/org/mule/runtime/core/api/config/MuleConfigurationTestCase.java:164]: Use allure annotations and @issue to refer to the corresponding ticket.

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java [hunk 6]
```diff
@@ -555,28 +553,11 @@ private void handleTxManagerFactory(TransactionManagerFactory object) {
...
-      getMuleRegistry().unregisterObject(OBJECT_SECURITY_MANAGER);
-      registerMuleSecurityManager = true;
-    } catch (Exception e) {
-      // NoSuchBeanDefinitionException can be ignored
-      if (!hasCause(e, NoSuchBeanDefinitionException.class)) {
-        throw new MuleRuntimeException(createStaticMessage("Error while unregistering Mule security manager"),
-                                       e);
-      }
-    }
-    if (registerMuleSecurityManager) {
```
Comment (fsgonz) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java:570]: Should this be removed?

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java (hunk 6 above)
Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java:570]: it is replaced by the proper remove* method calls

**Review by fsgonz — APPROVED** [review]

**Pull request merged**